src/utils.js

```diff
@@ -30,3 +30,12 @@ export const matches = (patterns, value) => {
 
   return patterns.some(pattern => values.some(val => pattern.test(val)))
 }
+
+export const machineDidUpdate = (oldMachine, newMachine) => {
```

Comment:
I think this should be:
`(prevMachineState.value !== machineState.value) || machineState.actions.length`

Otherwise, this would be true when the previous machine has actions, and the current one doesn't (i.e. it's invalid).

I would also replace `machine` with `state` or, better, with `machineState`, and rename it to something like `hasMachineStateChanged`.